src/common/vector_expression.hpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2019-2020, Lawrence Livermore National Security, LLC and
+// other Serac Project Developers. See the top-level LICENSE file for
+// details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/**
+ * @file vector_expression.hpp
+ *
+ * @brief A set of template classes used to represent the evaluation of unary and binary operations
+ * on vectors
+ */
+
+#ifndef VECTOR_EXPRESSION
+#define VECTOR_EXPRESSION
+
+#include "mfem.hpp"
+
+namespace serac {
+/**
+ * @brief A base class representing a vector expression
+ * @tparam T The base vector type, e.g., mfem::Vector, or another VectorExpr
+ * @note This class should never be used directly
+ */
+template <typename T>
+class VectorExpr {
+#ifdef NDEBUG
```

Comment:
Do we still need this?

Reply:
I spoke to @joshessman-llnl and after futher investigation this (and the note i mentioned that didn't get implemented) can be removed.  Even small performance hits should be taken for such a large readability/productivity boosts.  And its not even clear this has a performance hit overall.  Also this is a very small section of the total runtime.